packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -1638,6 +1638,19 @@ export default class NewVm extends BaseComponent {
 
   // ADVANCED --------------------------------------------------------------------
 
+  _getHost = createSelector(
+    () => this.props.pool,
+    () => this.props.pools,
+    () => this.props.template,
+    () => this.state.state.affinityHost,
+    (pool, pools, template, affinityHost) =>
+      affinityHost == null
+        ? pool === undefined
```

Comment:
The pool is only `undefined` in case of a self service user and the self service user don't have access to the pools collection.

Reply:
If the `resourceSet` contains the pool, we can get the `pool` objet, right?

Reply:
`CreateGetObjectOfType` checks the users' ACLs. A self service user don't has ACLs.